autospec/infile_bb_parser.py

```diff
@@ -0,0 +1,202 @@
+
+    # Split line to be [Key, operator, value] if in operators list
+    oper_pattern = r"(^[A-Z]+[_\-${}\[\]A-Za-z0-9]*)\s(" + '|'.join(
+        operators) + r")\s(\".*\")"
+
+    return re.compile(oper_pattern).search(line)
+
+
+def evaluate_expr(op, cur_val, assignment):
+    if not cur_val:
+        return assignment
+    return operators_dict.get(op).format(cur_val, assignment)
+
+
+def write_to_dict(bb_dict, m):
+
+    if len(m.groups()) == 3:
+        key = m.group(1)
+        value = clean_values(m.group(3))
+        op = m.group(2)
+
+        #
+        # ??= is the weakest variable assignment, so if that variable already
+        # has an assignment, do not overwrite it. = has the highest precedence
+        # and ?= is between the two.
+        #
+        if key in bb_dict and op == "??=":
+            return bb_dict
+        elif key in bb_dict and op == "?=":
+            if not isinstance(bb_dict[key], list):
+                return bb_dict
+
+        if key in bb_dict and isinstance(bb_dict[key], list):
+            v = bb_dict[key]
+            del [v[1]]
+            bb_dict[key] = "".join(v)
+        try:
+            bb_dict[key] = evaluate_expr(op, bb_dict.get(key), value)
+            if op == "??=":
+                bb_dict[key] = [bb_dict[key], 1]
+        except AttributeError as error:
+            print("Missing operation functionality: {}".format(error))
+
+    return bb_dict
+
+
+def bb_scraper(bb_fp, bb_dict):
+    todo = []
+
+    if bb_dict.get('filename'):
+        bb_dict = scrape_version(bb_dict)
+
+    for line in bb_fp:
+        cont = None
+        if line.strip() and not line.strip().startswith('#'):
+            line = line.strip()
+
+            # If line is a command, create raw string of the command
+            # TODO: command could be python code
+            if not cont and line.startswith('do_'):
+                cmd_name = line.split('()')[0].strip()
+                cmd = ["# Infile parser added the following lines: "]
+                depth = 0
+                while 1:
+                    depth = read_in_command(line, depth)
+                    if line:
+                        cmd.append("# " + line)
+
+                    if depth == 0:
+                        break
+                    else:
+                        line = next(bb_fp).strip()
+
+                if cmd[1].startswith("# " + cmd_name) and cmd[-1] == '# }':
+                    cmd = cmd[2:-1]
+
+                bb_dict[cmd_name] = cmd
+                continue
+            #
+            # # src_uri line continuation should be a list, not string
+            elif not cont and line[-1] == '\\':
+                lines = []
+                while 1:
+                    lines.append(line)
+                    line = next(bb_fp).strip()
+                    if line[-1] != '\\':
+                        lines.append(line)
+                        break
+
+                line = " ".join(lines)
```

Comment:
Above you say `src_uri line continuation should be a list, not a string`, but then you join the list into a string here.

Reply:
done